examples/Basic/console.py

```diff
@@ -0,0 +1,98 @@
+"""
+This example demonstrates how to make a graphical interface, and uses
+a random number generator to simulate data so that it does not require
+an instrument to use.
+
+Run the program by changing to the directory containing this file and calling:
+
+python gui.py
+
+"""
+
+import sys
+import random
+import tempfile
+from time import sleep
+
+from pymeasure.experiment import Procedure, IntegerParameter, Parameter, FloatParameter
+from pymeasure.experiment import Results
+from pymeasure.display.console import ManagedConsole
+from pymeasure.display.Qt import QtGui
+from pymeasure.display.windows import ManagedWindow
+import logging
+log = logging.getLogger('')
+log.addHandler(logging.NullHandler())
+
+
+class TestProcedure(Procedure):
+
+    iterations = IntegerParameter('Loop Iterations', default=100)
+    delay = FloatParameter('Delay Time', units='s', default=0.2)
+    seed = Parameter('Random Seed', default='12345')
+
+    DATA_COLUMNS = ['Iteration', 'Random Number']
+
+    def startup(self):
+        log.info("Setting up random number generator")
+        random.seed(self.seed)
+
+    def execute(self):
+        log.info("Starting to generate numbers")
+        for i in range(self.iterations):
+            data = {
+                'Iteration': i,
+                'Random Number': random.random()
+            }
+            log.debug("Produced numbers: %s" % data)
+            self.emit('results', data)
+            self.emit('progress', 100*i/self.iterations)
+            sleep(self.delay)
+            if self.should_stop():
+                log.warning("Catch stop command in procedure")
+                break
+
+    def shutdown(self):
+        log.info("Finished")
+
+
+class Console(ManagedConsole):
+
+    def __init__(self, argv):
+        super().__init__(
+            argv,
+            procedure_class=TestProcedure,
+            inputs=['iterations', 'delay', 'seed'],
+        )
+
+
+class MainWindow(ManagedWindow):
+
+    def __init__(self):
+        super(MainWindow, self).__init__(
+            procedure_class=TestProcedure,
+            inputs=['iterations', 'delay', 'seed'],
+            displays=['iterations', 'delay', 'seed'],
+            x_axis='Iteration',
+            y_axis='Random Number'
+        )
+        self.setWindowTitle('GUI Example')
+
+    def queue(self):
+        filename = tempfile.mktemp()
+
+        procedure = self.make_procedure()
+        results = Results(procedure, filename)
+        experiment = self.new_experiment(results)
+
+        self.manager.queue(experiment)
+
+
+if __name__ == "__main__":
+    if len(sys.argv) > 1:
+        app = Console(sys.argv)
+    else:
+        app = QtGui.QApplication(sys.argv)
```

Comment:
`QtGui.QApplication` is being deprecated, I think it should be replaced by:
```suggestion
        app = QtWidgets.QApplication(sys.argv)
```